packages/core/src/lib/metadata/index.ts

```diff
@@ -90,22 +96,39 @@ class Metadata extends BaseProtocol implements IMetadata {
       async (source) => await all(source)
     );
 
-    const decodedResponse = this.decodeMetadataResponse(encodedResponse);
+    const { error, shardInfo } = this.decodeMetadataResponse(encodedResponse);
+
+    if (error) {
```

Comment:
nit: this is a repeated line and should be moved into `decodeMetadataResponse` especially considering it is already doing `log.error` inside

Reply:
We still need the `error` state to be able to return it from the `query` function. Removed redundant logging

Reply:
971cd3c5ad0bc75927e820ff5a4e3e5b8148bcc5